Telemarketing condition

razvanudream opened the pull request:
#### What problem is this solving?

It helps display certain blocks (e.g. binding-selector) only for telemarketing users. This is needed because on sites where there are 2 or more bindings, logging into one of them will default the user to the first binding available on the site (login app doesn't take into account the sales channel / binding the user is on prior to logging in and thus causing problems for agents that want to place an order for a user on a certain binding)

#### How to test it?

1. Link this PR on a workspace
2. Add the Televendas access to your profile
3. Add this block to a store theme 

`"condition-layout.telemarketing#impersonable": {
    "props": {
      "conditions": [
        {
          "subject": "impersonable",
          "arguments": {
            "value": true
          }
        }
      ],
      "Then": "rich-text#show-component",
      "Else": "rich-text#binding-selector-hide"
    }
  },
  
  "rich-text#hide-component": {
    "props": {
      "text": "Not telesales agent, so do not show"
    }
  },
  
  "rich-text#show-component": {
    "props": {
      "text": "Telesales agent, show component"
    }
  }`

#### Screenshots or example usage:

![image](https://user-images.githubusercontent.com/71461884/143864408-14f21430-d888-42ef-b4b2-e170dd84641f.png)

#### Describe alternatives you've considered, if any.

I've considered adding a component that displays blocks based on a user role, but I couldn't find a graphql query that returns the user role based on a userId from the session.  Also, I've considered adjusting the login app to take into account the binding for .myvtex.com URLs but I don't have access to that and never will probably :)

#### How does this PR make you feel? [:link:](http://giphy.com/)

![](https://media.giphy.com/media/kiBcwEXegBTACmVOnE/giphy-downsized-large.gif)

mihainutiu-vtex commented:
Hi @igorbrasileiro and @victorhmp 
Can you please take a look at this PR. 
It is causing issues for a client that has a multi-binding architecture and needs to be able to get the correct sales channel in the checkout.

mihainutiu-vtex commented:
I published this beta version until this PR is approved and merged, @razvanudream 

`vtex.condition-layout@1.4.1-beta.0`

icazevedo commented on react/typings/vtex.render-runtime.d.ts:
```diff
@@ -0,0 +1,82 @@
+declare module 'vtex.render-runtime' {
```
I don't think these types are needed

icazevedo commented on react/hooks/useSession.ts:
```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react'
+import sessionQuery from 'vtex.store-resources/QuerySession'
+
+const useSession = (): Session | undefined => {
```
Instead of this hook, I think you can use [session-client](https://github.com/vtex-apps/session-client)'s useFullSession hook to get the data you want.

icazevedo commented:
Closed due to inactivity

razvanudream commented:
@icazevedo - thanks for your review.

I replaced the useSession hook with useFullSession from session-client app and removed react-apollo (as it's no longer needed), store-resources from manifest dependencies and typings that are no longer needed

mihainutiu-vtex commented:
Hi @icazevedo . 
The requested changes were resolved. Please review again.

Thank you

icazevedo reviewed (approved):
LGTM
